postgres/datadog_checks/postgres/relationsmanager.py

```diff
@@ -0,0 +1,186 @@
+ WHERE {relations}""",
+    'relation': True,
+}
+
+
+class RelationsManager(object):
+    def __init__(self, yamlconfig):
+        # type: (List[Union[str, Dict]]) -> None
+        self.log = get_check_logger()
+        self.config = self._build_relations_config(yamlconfig)
+
+    def build_relations_filter(self, schema_field):
+        # type (str) -> str
+        """Build a WHERE clause filtering relations based on relations_config."""
+        relations_filter = []
+
+        for r in self.config.values():
+            relation_filter = []
+            if r.get(RELATION_NAME):
+                relation_filter.append("( relname = '{}'".format(r[RELATION_NAME]))
+            elif r.get(RELATION_REGEX):
+                relation_filter.append("( relname ~ '{}'".format(r[RELATION_REGEX]))
+
+            if ALL_SCHEMAS not in r[SCHEMAS]:
+                schema_filter = ' ,'.join("'{}'".format(s) for s in r[SCHEMAS])
+                relation_filter.append('AND {} = ANY(array[{}]::text[])'.format(schema_field, schema_filter))
+
+            relation_filter.append(')')
+            relations_filter.append(' '.join(relation_filter))
+
+        return ' OR '.join(relations_filter)
+
+    def validate_relations_config(self, yamlconfig):
+        # type: (Dict) -> None
+        for element in yamlconfig:
+            if isinstance(element, dict):
+                if not (RELATION_NAME in element or RELATION_REGEX in element):
+                    raise ConfigurationError(
+                        "Parameter '%s' or '%s' is required for relation element %s",
+                        RELATION_NAME,
+                        RELATION_REGEX,
+                        element,
+                    )
+                if RELATION_NAME in element and RELATION_REGEX in element:
+                    raise ConfigurationError(
+                        "Expecting only of parameters '%s', '%s' for relation element %s",
+                        RELATION_NAME,
+                        RELATION_REGEX,
+                        element,
+                    )
+                if not isinstance(element.get(SCHEMAS, []), list):
+                    raise ConfigurationError("Expected '%s' to be a list for %s", SCHEMAS, element)
+            else:
+                raise ConfigurationError('Unhandled relations config type: %s', element)
+
+    def _build_relations_config(self, yamlconfig):
+        """Builds a dictionary from relations configuration while maintaining compatibility"""
+        config = {}
+        for element in yamlconfig:
+            if isinstance(element, str):
+                config[element] = {RELATION_NAME: element, SCHEMAS: [ALL_SCHEMAS]}
+            elif isinstance(element, dict):
+                relname = element.get(RELATION_NAME)
+                rel_regex = element.get(RELATION_REGEX)
+                schemas = element.get(SCHEMAS, [])
+                name = relname or rel_regex
+                config[name] = element.copy()
```

Comment:
Is there a risk we avoid thanks to this `copy` or is it to be safe?
Note that `schemas` here is still a reference to the original value, should we deepcopy it ?

Reply:
The code is just moved as it was from https://github.com/DataDog/integrations-core/pull/9322/files#diff-9113b2b59962ba80f2f82f3044b2e8eec514589c9c96f29f4582455d53098734L114